theia-extensions/viewer-prototype/src/browser/trace-viewer/trace-viewer.tsx

```diff
@@ -204,14 +205,21 @@ export class TraceViewerWidget extends ReactWidget {
                     } else {
                         const experiment = await this.experimentManager.openExperiment(this.uri.name + this.uri.ext, traces);
                         if (experiment) {
-                            this.openedExperiment = experiment;
-                            this.title.label = 'Trace: ' + experiment.name;
-                            this.id = experiment.UUID;
-
-                            if (this.isVisible) {
-                                signalManager().fireTraceViewerTabActivatedSignal(experiment);
+                            const widgets = this.widgetManager.getWidgets(TraceViewerWidget.ID);
+                            const widget = widgets.find(w => w.id === experiment.UUID);
+                            if (widget) {
+                                // Close widget if it had been opened previously.
+                                cancellation.cancel();
+                                this.dispose();
```

Comment:
It should make sure that other widget is brought to front.

Reply:
I'll update the patch to make sure it will.